lib/streamlit/elements/layouts.py

```diff
@@ -171,6 +171,93 @@ def column_proto(normalized_weight: float) -> BlockProto:
+
+        Inserts a number of multi-element containers as tabs.
+        Tabs are a navigational element that allows users to easily
+        move between groups of related content.
+
+        To add elements to the returned containers, you can use "with" notation
+        (preferred) or just call methods directly on the returned object. See
+        examples below.
+
+        .. warning::
+            All the content of every tab is always sent to and rendered on the frontend.
+            Conditional rendering is currently not supported.
+
+        Parameters
+        ----------
+        tabs : list of strings
+            Creates a tab for each string in the list. The string is used as the name of the tab.
+            The first tab is selected by default.
+
+        Returns
+        -------
+        list of containers
+            A list of container objects.
+
+        Examples
+        --------
+
+        You can use `with` notation to insert any element into a tab:
+
+        >>> tab1, tab2, tab3 = st.tabs(["Cat", "Dog", "Owl"])
+        >>>
+        >>> with tab1:
+        ...    st.header("A cat")
+        ...    st.image("https://static.streamlit.io/examples/cat.jpg")
+        ...
+        >>> with tab2:
+        ...    st.header("A dog")
+        ...    st.image("https://static.streamlit.io/examples/dog.jpg")
+        ...
+        >>> with tab3:
+        ...    st.header("An owl")
+        ...    st.image("https://static.streamlit.io/examples/owl.jpg")
+
+        .. output ::
+            https://share.streamlit.io/streamlit/docs/main/python/api-examples-source/layout.tabs1.py
+            height: 620px
+
+        Or you can just call methods directly in the returned objects:
+
+        >>> tab1, tab2 = st.tabs(["Chart", "Data"])
+        >>> data = np.random.randn(10, 1)
+        >>>
+        >>> tab1.subheader("A tab with a chart")
+        >>> tab1.line_chart(data)
+        >>>
+        >>> tab2.subheader("A tab with the data")
+        >>> tab2.write(data)
+
+
+        .. output ::
+            https://share.streamlit.io/streamlit/docs/main/python/api-examples-source/layout.tabs2.py
+            height: 750px
+
+        """
+        if not tabs:
+            raise StreamlitAPIException(
+                "The input argument to st.tabs must contain atleast one tab label."
+            )
+
+        if any(isinstance(tab, str) == False for tab in tabs):
+            raise StreamlitAPIException(
+                "The tabs input list to st.tabs is only allowed to contain strings."
+            )
+
+        def tab_proto(label: str) -> BlockProto:
+            tab_proto = BlockProto()
+            tab_proto.tab.label = label
+            tab_proto.allow_empty = True
+            return tab_proto
+
+        block_proto = BlockProto()
+        block_proto.tab_container.SetInParent()
+        tab_container = self.dg._block(block_proto)
+        return [tab_container._block(tab_proto(tab_label)) for tab_label in tabs]
```

Comment:
A minor nit:
Do you really need the return type to be mutable? If not, I'd prefer a tuple of DeltaGenerators as the return type.

Reply:
Good point. Since other similar elements (e.g. `columns`) already use lists as returns this is slightly more consistent. Refactoring other elements to use `tuple` is theoretically a breaking change, but in practice, there is probably nothing that will break. I'm fine with changing it to tuple here, but I'm not sure about that. @vdonato do you have an opinion on that?

Reply:
I changed it to tuple for now. But I used a sequence as typing hint so that people don't expect an actual tuple here.